### ISSUE TITLE
[med] Keyboard Navigation: Many Pages: Focus not moved/trapped within the opened dialog

### ISSUE THREAD

**Issue by ogumen** [Expensify/App#5516]
## Action Performed:
Prerequisite: the user is logged in with valid credentials.
1. Using Chrome, open https://staging.new.expensify.com/
2. Press Tab key to focus the button of user icon in the top left and press Enter key to activate it.
3. Once the dialog opens, note where the focus is moved.
4. Continue navigating by pressing Tab key and note if the focus is trapped within the dialog.

## Expected Result:
The focus should be moved to the opened dialog and correctly trapped within the dialog when navigating with Tab or Shift+Tab.

## Actual Result:
The focus is not moved to the dialog once it is opened. Also, the focus is not trapped in the dialog and moves to the background elements.

Other occurrences:
1. Request Money -> Upon opening the dialog, the focus is not trapped within it and moves to the background elements.
2. Split Bill -> Upon opening the dialog, the focus is not trapped within it and moves to the background elements.
3. Create a new chat overlay -> Upon opening the dialog, the focus is not trapped within it and moves to the background elements.
4. About overlay -> same issue is seen.
5. Similar issues are present for the following elements either focus not moving to at all or not the first focusable element (i.e. close, back, etc.): Not Moved --- Chat: - Request a call panel - Setting Panel - Add Attachment Settings: - Paypal.me Wrong Focus (not first element) --- - New Group Panel Chat: - Search panel - New chat FAB menu - New chat panel Settings - Add Phone Number

## Workaround:
No workaround

## Area issue was found in:
Settings overlay, Request money overlay, Split bill overlay, Create new chat overlay, About overlay, Request a call overlay, Chat page, Create new group overlay, Add payment method flow

## Failed WCAG checkpoints
2.4.3

## User impact:
The users might not understand the boundaries of the whole-screen dialog as focus is not set to it or trapped in it.

## Suggested resolution:
Make sure that the focus is set to the first actionable element in the dialog once it is opened and trapped in dialog until the user dismisses it. See: https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html and https://scottaohara.github.io/accessible_modal_window/

## Platform:
Where is this issue occurring?

- Web

**Version Number:**
**Reproducible in staging?:**Yes
**Reproducible in production?:**
**Notes/Photos/Videos:** Any additional supporting documentation
**Issue reported by:**

https://user-images.githubusercontent.com/88733897/134735043-750527d3-af6b-43e3-ba64-d81edd063eee.mp4

![Bug5182693_Focus_order_popup](https://user-images.githubusercontent.com/88733897/134735057-90a5a239-531d-4eda-85f8-aa0cdf740e30.png)
![Bug5182693_New_Chat-Focus_is_not_trapped_within_the_dialog](https://user-images.githubusercontent.com/88733897/134735065-cfb725a7-9063-45e3-92d9-594d02dce297.png)
![Bug5182693_Request_money-Focus_is_not_trapped_within_the_dialog](https://user-images.githubusercontent.com/88733897/134735074-923b7244-2a5c-4541-a037-87f5f6a25241.png)
![Bug5182693_Split_bill-Focus_is_not_trapped_within_the_dialog](https://user-images.githubusercontent.com/88733897/134735082-3b3a7271-07cf-4827-9b81-136f9bce6b1a.png)

https://user-images.githubusercontent.com/88733897/134735088-ab7a4e8c-609b-4787-95a0-d9a9a8dbbfcb.mp4

![Keyboard Navigation-Chat-Focus not moved to or trapped in dialogs   panels - add attachment](https://user-images.githubusercontent.com/88733897/134735101-c547280c-9ff0-4675-a260-c7660db57bd1.png)
![Keyboard Navigation-Chat-Focus not moved to or trapped in dialogs   panels - FAB menu](https://user-images.githubusercontent.com/88733897/134735105-e19ead47-3e69-487d-ab5a-3c81c85c11f5.png)
![Keyboard Navigation-Chat-Focus not moved to or trapped in dialogs   panels - request call](https://user-images.githubusercontent.com/88733897/134735109-8fcffa06-f205-4714-9669-83e2b2dca1bd.png)

https://user-images.githubusercontent.com/88733897/134735113-11452bd6-27ab-4d8a-ad8d-bda488b7330d.mp4

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
